plugin/githubinator.vim

```diff
@@ -28,31 +28,31 @@
 " SOFTWARE.
 
 " returns line numbers for selection
-function! GetRangeDelimiters() range
+function! s:get_range_delimiters()
     let l:beg = getpos("'<")[1]
     let l:end = getpos("'>")[1]
 
     return [l:beg, l:end]
 endfunction
 
 " get remote URL from .git/config
-function! s:generate_url() range
-    if !isdirectory('.git')
-        echoerr "githubinator: No .git directory found"
+function! s:generate_url()
+    let l:root_dir = system('git rev-parse --show-toplevel')
+    if v:shell_error != 0
+        echoerr printf('githubinator: %s is NOT managed by git', getcwd())
         return
     endif
 
-    let l:file_name = expand("%")
-    let l:beg = GetRangeDelimiters()[0]
-    let l:end = GetRangeDelimiters()[1]
-    let l:branch = system("git branch 2> /dev/null | awk '{print $2}' | tr -d '\n'")
-    let l:git_remote = system("cat .git/config | grep \"url\" | sed \"s/url.*://g\" | awk '{print \"https://github.com/\", $0}' | tr -d '[:blank:]' | sed \"s/\.git$//g\" | tr -d '\n'")
-    let l:final_url = l:git_remote . '/blob/' . l:branch . '/' . l:file_name . '#L' . l:beg . '-L' . l:end
+    " In order to support relative filepath, convert full path and remove the git root path in it.
+    let l:file_name = expand('%:p')[len(root_dir) : -1]
+    let [l:beg, l:end] = s:get_range_delimiters()
+    let l:branch = system("git name-rev --name-only HEAD | tr -d '\n'")
+    let l:remote = system("git config remote.origin.url | sed -Ee 's!(git@|git://)!https://!' -e 's!:([^/])!/\\1!' -e 's!\.git$!!' | tr -d '\n'")
```

Comment:
@mopp this is giving me `https://github.com/\1rakashdanish/vim-githubinator`, maybe change `'s!:([^/])!/\\1!'` to `'s!:([^/])!/\1!'`, you are actually escaping the `\` so it literally add `\1` to the string.

Reply:
Finally, I used substitute function of Vim for compatibility.